### PR TITLE
Fix primary-key generation and decimal metadata serialization in VectorStore.add

### PR DESCRIPTION
**Pull request by SasinduDilshara**
## Summary

Fixes the user-facing failure in [wso2/product-integrator#1100](https://github.com/wso2/product-integrator/issues/1100), where `ai:VectorKnowledgeBase.ingest` against a Milvus store crashed with `ConversionError - cannot convert '()' to type 'string'`, plus two adjacent issues found while building regression coverage for it.

> The decimal/JSON dynamic-field half of #1100 is fixed in the underlying Milvus connector — see [ballerina-platform/module-ballerinax-milvus#17](https://github.com/ballerina-platform/module-ballerinax-milvus/pull/17). This PR is the `ai.milvus` side: the primary-key fix that #1100 actually traces to, plus the consumer-side adjustments that let metadata flow through cleanly once the connector PR lands.

## Issue analysis

`ai:VectorKnowledgeBase.ingest` builds `ai:VectorEntry` values without populating `id`, expecting the underlying vector store to assign one. The previous implementation in `ballerina/vector_store.bal` did:

```ballerina
value: check int:fromString(check entry.id.cloneWithType())
```

That ran `cloneWithType` on a `nil` `id`, raising `{ballerina/lang.value}ConversionError - cannot convert '()' to type 'string'` — exactly the stack trace in #1100 (`vector_store.bal:98` → `rag.bal:147`). Every `ingest` call blew up before reaching Milvus.

While writing regression tests for that, two more problems surfaced in the metadata round-trip path:

1. `ai:NOT_IN` serialized as `"nin"` in the generated filter expression, but Milvus expects the literal `"not in"`. Every other operator already matched, so this was the only operator that broke filter queries.
2. `buildVectorMatch` was reading the dynamic `metadata` column as a string and running `jsondata:parseString` on it — a workaround that existed only because the connector used to stringify JSON columns. With [milvus#17](https://github.com/ballerina-platform/module-ballerinax-milvus/pull/17) the column now arrives as a structured `map<anydata>`, so the re-parse is both unnecessary and wrong (it loses type information).

### `ballerina/vector_store.bal`
- When `entry.id` is `()`, generate an INT64 primary key with `random:createIntInRange(1, int:MAX_VALUE)`. When the caller provides an `id` we still parse it as before, preserving existing behaviour for clients that manage their own keys. INT64 keeps the existing collection schema valid.

### `ballerina/utils.bal`
- Translate `ai:NOT_IN` → `"not in"` at the filter-string boundary; every other operator already matches Milvus' expected token, so the change is local to one branch.
- Drop the `jsondata:parseString` workaround in `buildVectorMatch`. Read `outputData["metadata"]` as `map<anydata>` directly and only attempt the `time:utcFromString` coercion on string-typed fields. Numbers, booleans, json arrays, and nested objects flow through with their types intact (this depends on milvus#17).
- Removed the now-unused `MetadataValues` record and `ballerina/data.jsondata` import.

### `ballerina/Ballerina.toml` / `Dependencies.toml`
- Added `ballerina/random` dependency to support the primary-key generation above.

### `ballerina/tests/test.bal` — regression coverage for #1100
Each test creates its own collection so they don't interfere with the existing `testInsert` / `testQuery` flow.

- `testInsertWithoutId` / `testQueryAutoGeneratedId` — covers the `ingest`-style path where `id` is omitted and asserts the entry is retrievable with a non-empty assigned id.
- `testInsertWithDecimalMetadata` / `testQueryWithDecimalMetadata` — asserts that `decimal` metadata fields round-trip and `cloneWithType(ai:Metadata)` succeeds (verifies the milvus#17 fix end-to-end through `ai.milvus`).
- `testInsertWithMixedJsonMetadata` / `testQueryWithMixedJsonMetadata` — broader coverage for nested objects, json arrays, floats, and booleans flowing through the open `json...` rest type.

### `ballerina/tests/test_advanced.bal` — new comprehensive suite
Sectioned coverage for the vector store, with one collection per test:
- **A. Init / configuration** — default config, custom primary key, custom chunk field, additional fields, error paths on bad configuration.
- **B. Data-type round-trip** — insert → query → verify for every supported metadata type.
- **C. Filter operators** — matching and non-matching cases for each operator, including the `NOT_IN` regression above.
- **D. Filter combinations** — AND / OR / nested compositions.
- **E. Edge cases and error paths**.
- **F. Custom configuration** — primary key, chunk field, additional fields.

## Test plan

- [ ] CI passes
- [ ] Run `bal test` locally against a local Milvus (`localhost:19530`) and confirm all new tests in `test.bal` and `test_advanced.bal` pass, with milvus#17 picked up via `Dependencies.toml`
- [ ] Verify `ai:VectorKnowledgeBase.ingest` against this store no longer raises `ConversionError`

### CODE REVIEW

**Review by Nuvindu — DISMISSED** [review]
LGTM

File: ballerina/utils.bal [hunk 1]
```diff
@@ -73,17 +75,19 @@ isolated function combineElements(string[] parts, string separator) returns stri
...
+        foreach string fieldName in rawMetadata.keys() {
+            anydata metadataValue = rawMetadata.get(fieldName);
+            if metadataValue is string {
+                time:Utc|time:Error utcFromString = time:utcFromString(metadataValue);
+                metadata[fieldName] = utcFromString is time:Utc ? utcFromString : metadataValue;
+            } else {
+                metadata[fieldName] = metadataValue;
+            }
         }
     }
```
[the remark this answers is not in the document]
Comment (SasinduDilshara) [reply ballerina/utils.bal:92]: This is ok for now.

**Review by Nuvindu — APPROVED** [review]
